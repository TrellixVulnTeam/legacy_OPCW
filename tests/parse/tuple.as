# Express automatic tuple types in a literal fashion.
();  # ok, empty tuple (man, so useless)
(231,);
(231, 21.21);
(32, false,);
(32, 42, 42, 3232e-1,);

# Declare various slots for automatic tuple types.
let q: (int, bool);
let w: (int, int,);
let e: (int, str, float64,);
let y: (uint,);

# Assign some tuples into those slots.
q = (32, true,);
w = (432, 32);
y = (135161,);

# # Declare and initialize various slots for automatic tuple types.
let z: (int,) = (324,);
let x: (int, int,) = (324, 12);
let c: (int, int, int, int) = (523, 23, 321, 12);
