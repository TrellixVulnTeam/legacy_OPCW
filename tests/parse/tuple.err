ModuleDecl <?> _
  Nodes <?> 
    TupleExpr <?>
      Elements <?> 
    TupleExpr <?>
      Elements <?> 
        IntegerExpr <?> 231 (10)
    TupleExpr <?>
      Elements <?> 
        IntegerExpr <?> 231 (10)
        FloatExpr <?> 21.21
    TupleExpr <?>
      Elements <?> 
        IntegerExpr <?> 32 (10)
        BooleanExpr <?> false
    TupleExpr <?>
      Elements <?> 
        IntegerExpr <?> 32 (10)
        IntegerExpr <?> 42 (10)
        IntegerExpr <?> 42 (10)
        FloatExpr <?> 3232e-1
    LocalSlotDecl <?> q
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
          Ident <?> bool
    LocalSlotDecl <?> w
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
          Ident <?> int
    LocalSlotDecl <?> e
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
          Ident <?> str
          Ident <?> float64
    LocalSlotDecl <?> y
      TupleExpr <?>
        Elements <?> 
          Ident <?> uint
    AssignExpr <?>
      Ident <?> q
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 32 (10)
          BooleanExpr <?> true
    AssignExpr <?>
      Ident <?> w
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 432 (10)
          IntegerExpr <?> 32 (10)
    AssignExpr <?>
      Ident <?> y
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 135161 (10)
    LocalSlotDecl <?> z
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 324 (10)
    LocalSlotDecl <?> x
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
          Ident <?> int
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 324 (10)
          IntegerExpr <?> 12 (10)
    LocalSlotDecl <?> c
      TupleExpr <?>
        Elements <?> 
          Ident <?> int
          Ident <?> int
          Ident <?> int
          Ident <?> int
      TupleExpr <?>
        Elements <?> 
          IntegerExpr <?> 523 (10)
          IntegerExpr <?> 23 (10)
          IntegerExpr <?> 321 (10)
          IntegerExpr <?> 12 (10)
