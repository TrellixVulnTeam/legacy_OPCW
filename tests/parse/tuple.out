ModuleDecl <?> _
  Nodes <?> 
    TupleExpr <?>
      Elements <?> 
    TupleExpr <?>
      Elements <?> 
        TupleExprMem <?>
          IntegerExpr <?> 231 (10)
    TupleExpr <?>
      Elements <?> 
        TupleExprMem <?>
          IntegerExpr <?> 231 (10)
        TupleExprMem <?>
          FloatExpr <?> 21.21
    TupleExpr <?>
      Elements <?> 
        TupleExprMem <?>
          IntegerExpr <?> 32 (10)
        TupleExprMem <?>
          BooleanExpr <?> false
    TupleExpr <?>
      Elements <?> 
        TupleExprMem <?>
          IntegerExpr <?> 32 (10)
        TupleExprMem <?>
          IntegerExpr <?> 42 (10)
        TupleExprMem <?>
          IntegerExpr <?> 42 (10)
        TupleExprMem <?>
          FloatExpr <?> 3232e-1
    SlotDecl <?> a
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?>
            Ident <?> int
          TupleTypeMem <?>
            Ident <?> bool
    SlotDecl <?> b
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?>
            Ident <?> int
          TupleTypeMem <?>
            Ident <?> int
    SlotDecl <?> c
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?>
            Ident <?> int
          TupleTypeMem <?>
            Ident <?> str
          TupleTypeMem <?>
            Ident <?> float64
    SlotDecl <?> d
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?>
            Ident <?> uint
    SlotDecl <?> e
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?> a
            Ident <?> int
          TupleTypeMem <?> b
            Ident <?> str
          TupleTypeMem <?> c
            Ident <?> float64
    SlotDecl <?> f
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?> a
            Ident <?> int
          TupleTypeMem <?> b
            Ident <?> uint
    SlotDecl <?> g
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?> a
            Ident <?> bool
    SlotDecl <?> h
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?> a
            Ident <?> bool
          TupleTypeMem <?>
            Ident <?> int
    SlotDecl <?> i
      TupleType <?>
        Elements <?> 
          TupleTypeMem <?>
            Ident <?> int
          TupleTypeMem <?> a
            Ident <?> bool
    AssignExpr <?>
      Ident <?> a
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 3214 (10)
          TupleExprMem <?>
            BooleanExpr <?> false
    AssignExpr <?>
      Ident <?> b
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 62 (10)
          TupleExprMem <?>
            IntegerExpr <?> 26 (10)
    AssignExpr <?>
      Ident <?> c
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 2351 (10)
          TupleExprMem <?>
            StringExpr <?> sdhs
          TupleExprMem <?>
            FloatExpr <?> 321.1672
    AssignExpr <?>
      Ident <?> d
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 2872 (10)
    AssignExpr <?>
      Ident <?> e
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            IntegerExpr <?> 26526 (10)
          TupleExprMem <?> b
            StringExpr <?> eysdgfh
          TupleExprMem <?> c
            FloatExpr <?> 236.213
    AssignExpr <?>
      Ident <?> e
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            Ident <?> a
          TupleExprMem <?> b
            Ident <?> b
          TupleExprMem <?> c
            FloatExpr <?> 21.21
    AssignExpr <?>
      Ident <?> f
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            IntegerExpr <?> 235 (10)
          TupleExprMem <?> b
            IntegerExpr <?> 235 (10)
    AssignExpr <?>
      Ident <?> g
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            BooleanExpr <?> false
    AssignExpr <?>
      Ident <?> g
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            Ident <?> a
    AssignExpr <?>
      Ident <?> h
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            BooleanExpr <?> false
          TupleExprMem <?>
            IntegerExpr <?> 51 (10)
    AssignExpr <?>
      Ident <?> h
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?> a
            Ident <?> a
          TupleExprMem <?>
            IntegerExpr <?> 51 (10)
    AssignExpr <?>
      Ident <?> i
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 734 (10)
          TupleExprMem <?> a
            BooleanExpr <?> true
    AssignExpr <?>
      Ident <?> i
      TupleExpr <?>
        Elements <?> 
          TupleExprMem <?>
            IntegerExpr <?> 231 (10)
          TupleExprMem <?> a
            Ident <?> a
